master/internal/db/postgres.go

```diff
@@ -885,61 +888,43 @@ WHERE id = :id`
 
 // DeleteExperiment deletes an existing experiment.
 func (db *PgDB) DeleteExperiment(id int) error {
-	tx, err := db.sql.Begin()
-	if err != nil {
-		return errors.Wrap(err, "error starting transaction")
-	}
-	defer func() {
-		if tx == nil {
-			return
+	return db.withTransaction("delete experiment", func(tx *sqlx.Tx) error {
```

Comment:
q: Is this relevant?

Reply:
nah i just made this code use `db.withTransaction` when i changed the sig to take a sqlx.Tx (since it was so easy to do).

Reply:
I was asking why deleting experiments relevant to supporting push metrics API. why do you need to change the code in this function?

Reply:
i'm confused, i answered that question above.

> nah i just made this code use db.withTransaction when i changed the sig to take a sqlx.Tx (since it was so easy to do).

nah is it not _strictly_ relevant but i just broke the signature of `db.deleteSnapshotsForExperiment` to require a `sqlx.Tx` not a `sql.Tx`so i moved it around to use `db.withTransaction` instead. you can checkout the code and change `withTransaction` to use a `sql.Tx` and you'll see why I made this change